src/bits/bit_array.rs

```diff
@@ -0,0 +1,181 @@
+use super::{BitArray, BooleanOps};
+use bitvec::prelude::{BitArr, Lsb0};
+use std::{
+    fmt::Debug,
+    ops::{BitAnd, BitAndAssign, BitOr, BitOrAssign, BitXor, BitXorAssign, Index, Not},
+};
+
+/// Bit store size of `XorSecretShare` in bytes.
+const BIT_STORE_SIZE_IN_BYTES: usize = 4;
+type U8_4 = BitArr!(for BIT_STORE_SIZE_IN_BYTES * 8, in u8, Lsb0);
+
+/// 32-bit array of bits. Similar to `u32`, but supports boolean algebra, and
+/// provides access to individual bits via index.
+///
+/// Bits are stored in the Little-Endian format. Accessing the first element
+/// like `s[0]` will return the LSB.
+///
+/// ## Example
+/// ```
+/// use raw_ipa::bits::BitArray32;
+///
+/// let s = BitArray32::from(2_u128);
+/// let b0 = s[0];
+/// let b1 = s[1];
+///
+/// assert_eq!(b0, 0);
+/// assert_eq!(b1, 1);
+/// ```
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub struct BitArray32(U8_4);
+
+impl BitArray for BitArray32 {
+    const SIZE_IN_BYTES: usize = BIT_STORE_SIZE_IN_BYTES;
+
+    #[allow(dead_code)]
+    const ZERO: Self = Self(U8_4::ZERO);
```

Comment:
I tried creating a `const ONE`, but couldn't do it without using a macro or ::new(). I think we are okay without it though.

Reply:
it is a bit ugly but works and with compile-time checks that value fits into 4 byte slice

```rust
    const ONE: Self = BitArray32(bitarr![const u8, Lsb0; 1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]);
```

Reply:
I don't see why we would need a definition for ONE here.  There might be value in `~ZERO` though because from that you can get (`~ZERO >> n`).  Of course, it's easy to produce that as needed without the constant.